### PR TITLE
Upload Attached File: Idea

### PR DESCRIPTION
**Pull request by stedmanblake**
### Summary
- Upload file attachments in `ideas/form`
- Remove file attachments in `ideas/form`
- Show file attachments in `ideas/show`
- Not showing attachments in `ideas/card/show` since we neither show links. Happy to update both if you think that'd look better (a la `experiences/card/show`: 
  image)
### Screenshots
- Upload file attachments in `ideas/form`
  image
- Remove file attachments in `ideas/form`
  image
- Show file attachments in `ideas/show`
  image
- Not showing attachments in `ideas/card/show` since we neither show links. Happy to update both if you think that'd look better
  image

### CODE REVIEW

**Comment by stedmanblake** [comment]
@krnjn i'm making separate PRs for file attachments on `idea` and `recipe`.

If you make changes to these before merging, can you keep the commits separate? In PRs 1 and 2, my commits were rolled up into yours, which made it hard to parse what I'd done v. what you'd done when reviewing your changes. Any idea how that happened / how we can avoid that in the future? Not sure if it was something wonky on my end.

**Comment by krnjn** [comment]
Hey man – sorry about the squashing commits, it's how we do things at my current job so got into a bad habit of doing that. No worries on this project. One thing that I would like to do if possible though is to avoid introducing any of the other seeds files into these PRs here – I updated master so that it has a bunch of fake user / challenge data that should work for purposes of local development, but some of these still have the seeds data in the PR, so would be best to remove it if possible.

Another way to get the seeds data onto your branch from master is to:

`git pull origin master` (presuming you haven't made any commits on this
`git checkout branchname` (do all your changes on this branch)
`git rebase master` (this basically applies all of master first, then all of your changes)
`git commit --amend` (will amend the changes to your commit)
`git push -f` (will push up the changes to this branch)

Let me know if that doesn't work and we can try to figure out something else about how to get those seeds on your branch so you can use this. But generally I think this is looking pretty good!

**Comment by stedmanblake** [comment]
@krnjn OK, I've ironed out all of that mess -- so much reverting! I'll definitely make sure to avoid that in the future. Thanks for your help with it.

This is now ready for your review!

**Comment by krnjn** [comment]
This looks good – just add the `file` arg to the idea_controller_spec (as was done in PR #12 for experiences) and we should be good to go!

**Comment by stedmanblake** [comment]
Right on! Back over to you
